tools/hardware/outputs/json.py

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright 2023, UNSW
+#
+# SPDX-License-Identifier: GPL-2.0-only
+#
+
+''' Generate a JSON file with memory region info from the device tree '''
+
+import argparse
+import json
+from typing import List
+import hardware
+from hardware.config import Config
+from hardware.fdt import FdtParser
+from hardware.utils.rule import HardwareYaml
+
+
+def make_json_list_of_regions(regions) -> List:
+    # Output each address as a hex-string (e.g "0x8000000")
+    return [
+        {
+            'start': f"0x{r.base:x}",
```

Comment:
There is a JSONEncoder class thing for doing this conversion systematically, since the default encoding is Python `int` to JSON numbers. Given the simplicity of this configuration file I have done it manually here, as I think that this approach is simpler.

Reply:
Can you extend the comment, that we don't use JSON-integers but hex-strings, because we prefer readability by humans. And we deeply regret the current comment JSON spec lacks hex integer support.